contracts/os/andromeda-kernel/src/execute.rs

```diff
@@ -797,23 +798,68 @@ impl MsgHandler {
                 error: Some(format!("Channel not found for chain {chain}")),
             });
         }?;
-        let ctx = ctx.map_or(
-            AMPPkt::new(
-                info.sender,
-                env.clone().contract.address,
-                vec![AMPMsg::new(
-                    recipient.clone().get_raw_path(),
-                    message.clone(),
-                    None,
-                )],
-            ),
+        let ctx = ctx.map_or_else(
+            || {
+                // Check if sender has username
+                let vfs_address = KERNEL_ADDRESSES.load(deps.storage, VFS_KEY).unwrap();
+                let msg = VFSQueryMsg::GetUsername {
+                    address: info.sender.clone(),
+                };
+                let username_addr =
+                    deps.querier
+                        .query::<String>(&cosmwasm_std::QueryRequest::Wasm(
+                            cosmwasm_std::WasmQuery::Smart {
+                                contract_addr: vfs_address.to_string(),
+                                msg: to_json_binary(&msg).unwrap(),
+                            },
+                        ));
```

Comment:
Can we not use a raw query for this using `AOSQuerier`?

Reply:
If not we should separate this closure in to a separate function as it's quite lengthy.